libenv/sysinfo.c

```diff
@@ -3335,6 +3335,93 @@ void GetDefVars(EvalContext *ctx)
+                                      "source=agent,derived=debian");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "centos") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "CentOS", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=centos");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "fedora") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Fedora", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=fedora");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "redhat") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "RHEL", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=redhat");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "aix") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "AIX", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=aix");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "hpux") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "HP-UX", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=hpux");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "suse") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "SUSE", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=suse");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "macos") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "macOS", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=macos");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "windows") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Windows", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=windows");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "freebsd") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "FreeBSD", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=freebsd");
+    }
+    else if (EvalContextClassGet(ctx, NULL, "solaris") != NULL)
+    {
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
+                                      "Solaris", CF_DATA_TYPE_STRING,
+                                      "source=agent,derived=windows");
+    }
```

Comment:
I think we should also have a default like *Unknown*. It's better than not expanding the variable when it's attempted to be used and doesn't exist.

Reply:
Yes, that makes sense